025-control-plain-listener.md

```diff
@@ -0,0 +1,80 @@
+# Control Plane Listener
+
+This proposal suggests to use separate Control Plane listener and the plan how it should be introduced.
+
+## Current situation
+
+Strimzi currently has one internal listener.
+It is used by the Kafka brokers for data replication but also for coordination between the controller and the other brokers.
+And in addition to that, it is also used by the different Strimzi components:
+* Operators
+* Kafka Exporter
+* Cruise Control
+
+## Motivation
+
+Kafka already for some time supports using separate listeners for data replication and coordination.
+The main reason for using separate listeners is that the replication traffic which is very data-intensive does not increase the latency the coordination traffic when separate listeners are used.
+Strimzi should make use of it and have separate listeners for the different tasks.
+
+## Proposal
+
+Introducing the separate control plain listener has impact on backwards compatibility and on upgrade / downgrade procedures.
+During the upgrade or downgrade, different brokers which are part of the same cluster would have a different configuration.
+Some of them will try to coordinate using the dedicated control plane listener while others will try to coordinate using the replication listener.
+Because of this, the different members of the cluster will not be able to communicate and the cluster will not be available to clients during the upgrade or downgrade.
+This can be easily handled while upgrading.
+But it is not possible to handle it when downgrading as the version the user would be downgrading from will not have a chance to revert the control plane listener changes.
+
+To mitigate this, this proposal suggests to use [Feature Gates](https://github.com/strimzi/proposals/blob/main/022-feature-gates.md) to introduce the control plane listener over multiple releases and cause minimal disruption for the users.
+Strimzi 0.23 will add new internal listener using port 9090.
+This listener will be enabled by default and configured in the same way as the replication listener.
+But it will not be configured as the control plane listener.
+
+A new feature gate `ControlPlaneListener` will be added and disabled by default.
+When enabled, this feature gate will configure the Kafka brokers to use the new listener for the control plane communication.
```

Comment:
Does "the control plane communication" include the connections made by the CO to the brokers?

Reply:
The plan expects them to keep using the replication listener. The Control Plane is only the Kafka control plane here. I updated the text to make it more clear.